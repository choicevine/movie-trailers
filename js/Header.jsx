// @flow
import React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  searchInput?: boolean,
  handleSearchTerm?: Function
};
const Header = (props: Props) => {
  let utilSpace;

  if (props.searchInput) {
    utilSpace = <input type="text" onChange={props.handleSearchTerm} placeholder="search" />;
  } else {
    utilSpace = (
      <h2>
        <Link to="/search">Back</Link>
      </h2>
    );
  }

  return (
    <header>
      <h1>
        <Link to="/">Svideos</Link>
      </h1>
      {utilSpace}
    </header>
  );
};
Header.defaultProps = {
  searchInput: false,
  handleSearchTerm: function noop() {}
};

export default Header;
