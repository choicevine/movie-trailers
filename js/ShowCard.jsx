// @flow
import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
	width: 32%;
	border: 2px solid #333
	border-radius: 4px;
	margin-bottom: 25px;
	padding-right: 10px;
	overflow: hidden;
`;

type Props = {
  title: string,
  description: string,
  year: string,
  poster: string
};

const ShowCard = (props: Props) => (
  <Wrapper className="show-card">
    <img alt={`${props.title} show poster`} src={`/public/img/posters/${props.poster}`} />
    <div>
      <h3>{props.title}</h3>
      <h4>{props.year}</h4>
      <p>{props.description}</p>
    </div>
  </Wrapper>
);
export default ShowCard;
