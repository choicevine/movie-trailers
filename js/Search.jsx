// @flow

import React, { Component } from 'react';
import ShowCard from './ShowCard';
import Header from './Header';


type State = {
  searchTerm: string
};

type Props = {
  preload: Array<{
    title: string,
    id: string,
    description: string,
    year: string,
    poster: string
  }>
};

class Search extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      searchTerm: ''
    };
    //  this.handleSearchTerm = this.handleSearchTerm.bind(this);
  }
  handleSearchTerm = (event: SyntheticKeyboardEvent<HTMLInputElement>) => {
    this.setState({ searchTerm: event.currentTarget.value });
  };
  render() {
    return (
      <div className="search">
        <Header searchInput handleSearchTerm={this.handleSearchTerm} />
        <div>
          {this.props.preload
            .filter(
              show =>
                `${show.title} ${show.description}`.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0
            )
            .map(show => <ShowCard key={show.imdbID} {...show} />)}
        </div>
      </div>
    );
  }
}

export default Search;
