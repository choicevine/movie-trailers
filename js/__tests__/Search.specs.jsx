import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter } from 'react-router';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from '../Search';
import Header from '../Header';
import ShowCard from '../ShowCard';
import preload from '../../data.json';

Enzyme.configure({ adapter: new Adapter() });

test('Search renders correctly', () => {
  const data = [{ imdbID: 'abc', title: 'billion' }, { imdbID: 'ayuc', title: 'shepeteri' }];

  const component = shallow(<Search preload={data} />);

  expect(component).toMatchSnapshot();
});

test('Search Should render correct amount of Showcard Components', () => {
  const data = [{ imdbID: 'abc', title: 'billion' }, { imdbID: 'ayuc', title: 'shepeteri' }];

  const component = shallow(<Search preload={data} />);

  expect(data.length).toEqual(component.find(ShowCard).length);
});

test('Should render correct amount of ShowCard component for search field ', () => {
  const searchWord = 'black';
  const component = shallow(<Search preload={preload.shows} />);
  const headerComponent = shallow(<Header searchInput />);

  headerComponent.find('input').simulate('change', { target: { value: searchWord } });

  const showCount = preload.shows.filter(show =>
    `${show.title} ${show.description}`.toUpperCase().indexOf(searchWord.toUpperCase() >= 0)
  );

  expect(showCount.length).toEqual(component.find(ShowCard).length);
});
