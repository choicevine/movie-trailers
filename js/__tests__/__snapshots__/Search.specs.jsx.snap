// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <Header
    handleSearchTerm={[Function]}
    searchInput={true}
  />
  <div>
    <ShowCard
      imdbID="abc"
      key="abc"
      title="billion"
    />
    <ShowCard
      imdbID="ayuc"
      key="ayuc"
      title="shepeteri"
    />
  </div>
</div>
`;
